src/usr/local/www/vpn_openvpn_client.php

```diff
@@ -634,6 +634,13 @@
 		$pconfig['auth_pass']
 	))->setHelp('Leave empty when no password is needed');
 
+	$section->addInput(new Form_Textarea(
+		'auth_script',
+		'Auth Script',
+		$pconfig['auth_script']
+	))->setHelp('Script that would output credentials as "user<newline>password".' .
+	    'When defined, this will be executed and the results used instead of username and password. Allows use of OTP authentication.', '<br/>');
```

Comment:
If this is intended to be a full path to a script on the filesystem, that should be mentioned in the help text.

Input validation must also ensure this is a path to a script file alone and not a string of other commands or parameters, otherwise you're handing the user a way to exec anything they want in the shell.

This should be more clear about it being executed when the settings are saved and not every time the client connects. Given that limitation I'm not sure how well it will effectively work with OTP since if the client gets disconnected you would have to re-save the settings or otherwise trigger an openvpn config rewrite to get a new token in the file.